### ISSUE TITLE
Focus on datepicker could open datepicker

### ISSUE THREAD

**Issue by franckstifler** [vmware/clarity#4367]
## Summary

Provide a means above clicking on the calendar icon, to trigger up the picker, this could facilitate UX on the date picker.
I'm thinking about having an event like on the material datepicker that helps fire up the picker.

## Use case
Filling a form using the tab key to switch fields, the user needs to click on the calendar icon to popup the picker.

## Workarounds

Tried selecting the button that follows the focused input, and triggered a click event on the button element in the component.ts file, but the behaviour is not stable and throws errors sometimes in the console.

**Comment by gnomeontherun** [comment]
I think from an accessibility standpoint, that is potentially confusing. The focus would start on the input, but then get moved before reading the input. Keyboard users would not get to hear the current value before being forced into a new place, which makes it harder to understand what is happening. 

@chlane do you have any guidance on if this is an accessible pattern?

**Comment by chlane** [comment]
I agree with @gnomeontherun, since screen reader cursors follow browser focus, if focus is moved involuntarily, blind screen reader users may not be able to read some content, like the label for the input field or any other content  from aria-labelledby or aria-describedby. Keyboard-only and blind screen reader users should be able to read and activate controls at their own pace and discretion by using the Tab key to focus controls and the Enter key to activate them.

**Comment by franckstifler** [comment]
@gnomeontherun and @chlane the idea is not to enable this by default, but instead to provide a means whenever needed to open the picker, without only clicking on the icon. As I said, it's also available on mat-datepicker (on which you can call a open method to popup the picker).

**Comment by chlane** [comment]
But if the date picker can only be opened by clicking then how is a user who is not using a mouse going to open it?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by franckstifler** [comment]
@chlane I don't know if it's to me you're addressing, in any case, I'm requesting an event on the datepicker, that we could invoke maybe on a focus, keyup... event.

**Comment by chlane** [comment]
@franckstifler Hi franckstifler. Opening the datepicker when its trigger gains focus will cause an involuntary movement of focus. When the datepicker opens, the best practice is to move focus inside of it. Due to this, It is important to let sighted keyboard-only and bind screen reader users initiate the appearance of controls like datepickers and dialogs by activating a focused trigger element with the Enter key. Additionally, users should be able to tab past the date picker's triggering element without triggering the appearance of the date picker. Low-vision keyboard-only users and blind screen reader users may try to explore the controls on the page using the Tab key and be confused by having to interact with a date picker control that they do not intend to interact with. 

Even if it is not enabled by default, creating an inaccessible feature contradicts our mission of ensuring  our products are born accessible and  introduces more complexity to accessibility testing and conformance reporting (ACRs/VPATs).

**Comment by gnomeontherun** [comment]
I'm going to close this as something we won't do, since it has accessibility implications. Thanks for the suggestion though, I understand the request.
